File: backend/src/models/Interview.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IInterview extends Document {
  userId: string;
  sessionId: string;
  date: Date;
  score: number;
  feedback: string;
  verbatim: { role: string; content: string }[];
  improvements: string[];
  verdict: string;
  difficulty?: string;
  questionsAttempted?: number;
  cheatingFlags?: string[];
  tabSwitches?: number;
}

const InterviewSchema: Schema = new Schema(
  {
    userId:             { type: String, required: true, index: true },
    sessionId:          { type: String, required: true, unique: true, index: true },
    date:               { type: Date, default: Date.now },
    score:              { type: Number, required: true },
    feedback:           { type: String, required: true },
    verbatim:           [{ role: { type: String }, content: { type: String }, _id: false }],
    improvements:       [{ type: String }],
    verdict:            { type: String, default: 'Pending' },
    difficulty:         { type: String, enum: ['easy', 'medium', 'hard'], default: 'medium' },
    questionsAttempted: { type: Number, default: 0 },
    cheatingFlags:      [{ type: String }],
    tabSwitches:        { type: Number, default: 0 },
  },
  {
    timestamps: true,
  }
);

// Compound index for efficient history queries (sorted by date)
InterviewSchema.index({ userId: 1, date: -1 });

export default mongoose.models.Interview ||
  mongoose.model<IInterview>('Interview', InterviewSchema);

File: backend/src/main/java/com/aiinterview/model/Interview.java
package com.aiinterview.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.index.CompoundIndex;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "interviews")
@CompoundIndex(name = "userId_date_idx", def = "{'userId': 1, 'date': -1}")
public class Interview {

    @Id
    private String id;

    @Indexed
    private String userId;

    @Indexed(unique = true)
    private String sessionId;

    private Instant date;

    private int score;

    private String feedback;

    // Transcript: [{role, content}, ...]
    private List<Map<String, String>> verbatim;

    private List<String> improvements;

    @Builder.Default
    private String verdict = "Pending";

    // "easy" | "medium" | "hard"
    private String difficulty;

    @Builder.Default
    private int questionsAttempted = 0;

    private List<String> cheatingFlags;

    @Builder.Default
    private int tabSwitches = 0;

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}
